spring-kafka/src/main/java/org/springframework/kafka/config/RetryTopicComponentFactory.java

```diff
@@ -0,0 +1,200 @@
+	 * {@link DestinationTopic}.
+	 * @param destinationTopicResolver the {@link DestinationTopicResolver} instance to
+	 * resolve the destinations.
+	 * @return the instance.
+	 */
+	public DeadLetterPublishingRecovererFactory deadLetterPublishingRecovererFactory(
+			DestinationTopicResolver destinationTopicResolver) {
+
+		return new DeadLetterPublishingRecovererFactory(destinationTopicResolver);
+	}
+
+	/**
+	 * Create the {@link ListenerContainerFactoryResolver} that will be used to resolve
+	 * the appropriate {@link KafkaListenerContainerFactory} for a given topic.
+	 * @param beanFactory the {@link BeanFactory} that will be used to retrieve the
+	 * {@link KafkaListenerContainerFactory} instance if necessary.
+	 * @return the instance.
+	 */
+	public ListenerContainerFactoryResolver listenerContainerFactoryResolver(BeanFactory beanFactory) {
+		return new ListenerContainerFactoryResolver(beanFactory);
+	}
+
+	/**
+	 * Create a {@link ListenerContainerFactoryConfigurer} that will be used to
+	 * configure the {@link KafkaListenerContainerFactory} resolved by the
+	 * {@link ListenerContainerFactoryResolver}.
+	 * @param kafkaConsumerBackoffManager the {@link KafkaConsumerBackoffManager} used
+	 * with the {@link KafkaBackoffAwareMessageListenerAdapter}.
+	 * @param deadLetterPublishingRecovererFactory the factory that will provide the
+	 * {@link DeadLetterPublishingRecoverer} instance to be used.
+	 * @param clock the {@link Clock} instance to be used with the listener adapter.
+	 * @return the instance.
+	 */
+	public ListenerContainerFactoryConfigurer listenerContainerFactoryConfigurer(KafkaConsumerBackoffManager kafkaConsumerBackoffManager,
+			DeadLetterPublishingRecovererFactory deadLetterPublishingRecovererFactory,
+			Clock clock) {
+
+		return new ListenerContainerFactoryConfigurer(kafkaConsumerBackoffManager, deadLetterPublishingRecovererFactory, clock);
+	}
+
+	/**
+	 * Create the {@link RetryTopicNamesProviderFactory} instance that will be used
+	 * to provide the property names for the retry topics' {@link KafkaListenerEndpoint}.
+	 * @return the instance.
+	 */
+	public RetryTopicNamesProviderFactory retryTopicNamesProviderFactory() {
+		return new SuffixingRetryTopicNamesProviderFactory();
+	}
+
+	/**
+	 * Create the {@link KafkaBackOffManagerFactory} that will be used to create the
+	 * {@link KafkaConsumerBackoffManager} instance used to back off the partitions.
+	 * @param registry the {@link ListenerContainerRegistry} used to fetch the
+	 * {@link MessageListenerContainer}.
+	 * @return the instance.
+	 */
+	public KafkaBackOffManagerFactory kafkaBackOffManagerFactory(ListenerContainerRegistry registry) {
+		return new PartitionPausingBackOffManagerFactory(registry);
+	}
+
+	/**
+	 * Create the {@link TaskExecutor} that will be used in the
+	 * {@link KafkaConsumerTimingAdjuster}.
+	 * @return the task executor.
+	 */
+	public TaskExecutor taskExecutor() {
+		return new ThreadPoolTaskExecutor();
```

Comment:
Don't we provide a bean now?
What is the point of this factory method if end-user will have to just provide a respective bean to override a default one?